pkgs/os-specific/linux/bolt/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, meson, ninja, pkgconfig, fetchFromGitLab,
+  python3, umockdev, gobject-introspection,
+  git, asciidoc, libxml2, libxslt, docbook_xml_dtd_45, docbook_xsl,
+  glib, systemd, polkit
+}:
+
+stdenv.mkDerivation rec {
+  name = "bolt-${version}";
+  version = "0.7";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "bolt";
+    repo = "bolt";
+    rev = "${version}";
+    sha256 = "0xn2c31kcjh1j76gq1qrcxwjyjyqnsxygkfrvh3xk07qc92f99xd";
+  };
+
+  nativeBuildInputs = [
+    meson ninja pkgconfig umockdev
+    git asciidoc libxml2 libxslt docbook_xml_dtd_45 docbook_xsl
+    (python3.withPackages
```

Comment:
Maybe these should go to `checkInputs`?

Reply:
Doesn't work unfortunately. Bails out on the configurePhase:
> Program /build/source/tests/test-integration found: YES (/build/source/tests/test-integration)
> 
> meson.build:422:0: ERROR:  Could not execute command "/build/source/tests/test-integration list-tests".

Reply:
Even with `doCheck = true`?

Reply:
Ah, that fixes the configurePhase (thought `doCheck` was true by default :confused: ), but the tests apparently fail, looks they actually didn't run before.

Reply:
The tests require `dbus` and we'll need to patch paths for [`LD_PRELOAD`](https://gitlab.freedesktop.org/search?utf8=%E2%9C%93&search=ld_preload&group_id=&project_id=32&search_code=true&repository_ref=master#L416) at least.

Reply:
Will not `umockdev` be picked up from `checkInputs`?

Reply:
Hmm, might be another problem:
> ERROR: ld.so: object 'libumockdev-preload.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libumockdev-preload.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libumockdev-preload.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
test_device_auto_import_sl2 (__main__.BoltTest) ...
** (process:3779): WARNING **: 13:29:58.068: Failed to load shared library 'libumockdev.so.0' referenced by the typelib: libumockdev.so.0: cannot open shared object file: No such file or directory

Reply:
Right, `umockdev` uses vala, so we're hit by the broken gir problem.

Reply:
(I'm out of my depth on resolving this one; if I can be of assistance, let me know, otherwise I'll observe and learn.)

Reply:
That should not really be an issue if we add it to `LD_LIBRARY_PATH`.

Reply:
Yeah, adding absolute paths to `LD_PRELOAD` and adding `${umocdev.out}/lib` to `LD_LIBRARY_PATH` fixed the problem. (note the store test fails without 462a6d32501b204944a9ef3f04a6d28f28c36cbc)

Reply:
Right, and `LD_LIBRARY_PATH` makes the `LD_PRELOAD` fix unnecessary of course.

so this should do the trick:
```
  checkInputs = [
    umockdev dbus gobject-introspection
    (python3.withPackages
      (p: [ p.pygobject3 p.dbus-python p.python-dbusmock ]))
  ];
  doCheck = true;

  preCheck = ''
    export LD_LIBRARY_PATH=${umockdev.out}/lib/
  '';
```

Reply:
That works, but it still fails if `doCheck = false` because it invokes  What should we do about that?

Options I can imagine:

1. Change Python to a nativeBuildInput? (and maybe set withPackages conditionally on doCheck?)

2. Conditionally disable test/test-integration based on `doCheck`

Or something else entirely

Reply:
Yeah, we tend to include tests in the build when possibly, so `doCheck = true;` should be there. I guess we could throw `python3` in nativeBuildInputs too in case someone disables checking, but I don't quite see the point as the tests are fast.

Reply:
Done. I kept `doCheck = true;` and added a conditional to append `python3` to `nativeBuildInputs` when needed.